File: Assets/Scripts/Interfaces/IMap.cs
using UnityEngine;
using System.Collections;

public interface IMap {
	INavigator Navigator { get; }
	int NumRows { get; }
	int NumCols { get; }
	GameObject this[int row, int col] { get; }
	void Generate();
	void Reset();
	CellType GetCellType(int row, int col);
	GameObject GetStartCell(out int startRow, out int startCol);
}

File: Assets/Scripts/Interfaces/IMap.js
#pragma strict

public interface IMap {
	function Navigator(): INavigator;
	function NumRows(): int;
	function NumCols(): int;
	function Map(row: int, col: int): GameObject;
	function Generate();
	function Reset();
	function GetCellType(row: int, col: int): CellType;
	function GetStartCellIndex(): CellIndex;
	function PlaceItemInRandomRoom(item: Transform);
	function ValidMapCell(row: int, col: int): boolean;
}